collator/src/lib.rs

```diff
@@ -273,8 +273,11 @@ impl<P: 'static, SP: 'static> RelayChainContext for ApiContext<P, SP> where
 	}
 }
 
-/// Run the collator node using the given `service`.
-fn run_collator_node<S, P, Extrinsic>(
+/// Build a collator service with the given `RelayChainContext` and `ParachainContext`
+/// built by the given `BuildParachainContext` and arguments to the underlying polkadot node.
+///
+/// This function does not block. It returns an `AbstractService` that can be used as a future
+pub fn build_collator_service<S, P, Extrinsic>(
```

Comment:
Why is that required to be public?

Reply:
I thought it was used by the cli but actually it is not.